File: src/main/java/jp/aoichaan0513/A_TosoGame_Live/API/Manager/World/WorldManager.java
package jp.aoichaan0513.A_TosoGame_Live.API.Manager.World;

import jp.aoichaan0513.A_TosoGame_Live.API.Manager.GameManager;
import jp.aoichaan0513.A_TosoGame_Live.Main;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.World;

public class WorldManager {

    public static World getWorld() {
        return Bukkit.getWorld(getWorldName());
    }

    public static String getWorldName() {
        return Main.getMainConfig().getString("worldName");
    }

    public static void setWorld(World world) {
        if (GameManager.isGame()) return;
        Main.getMainConfig().set("worldName", world.getName());
        Main.setMainConfig();
        Main.getInstance().reloadConfig();
        return;
    }

    public static void setWorld(String world) {
        if (GameManager.isGame()) return;
        Main.getMainConfig().set("worldName", world);
        Main.setMainConfig();
        Main.getInstance().reloadConfig();
        return;
    }

    public static void setWorld(World world, boolean bool) {
        if (GameManager.isGame()) return;
        if (bool) {
            Main.getMainConfig().set("worldName", world.getName());
            Main.setMainConfig();
            return;
        } else {
            setWorld(world);
            return;
        }
    }

    public static void setWorld(String world, boolean bool) {
        if (GameManager.isGame()) return;
        if (bool) {
            Main.getMainConfig().set("worldName", world);
            Main.setMainConfig();
            return;
        } else {
            setWorld(world);
            return;
        }
    }

    public enum Difficulty {
        EASY(ChatColor.GREEN + "イージー"),
        NORMAL(ChatColor.YELLOW + "ノーマル"),
        HARD(ChatColor.RED + "ハード"),
        HARDCORE(ChatColor.DARK_RED + "ハードコア");

        private final String displayName;

        private Difficulty(String displayName) {
            this.displayName = displayName;
        }

        public String getName() {
            return name().toLowerCase();
        }

        public String getDisplayName() {
            return displayName;
        }

        public static Difficulty getDifficulty(String name) {
            return valueOf(name.toUpperCase());
        }
    }

    // ゲームタイプ
    public enum GameType {
        START,
        RESPAWN
    }


    public enum PathType {
        MAP("map"),
        GAME_TIME("game.time"),
        GAME_OTHER("game.other"),

        OPGAME_DICE("opgame.dice"),

        LOCATION_OPGAME("location.opgame"),
        LOCATION_GOPGAME("location.gopgame"),
        LOCATION_HUNTER("location.hunter"),
        LOCATION_JAIL("location.jail"),
        LOCATION_RESPAWN("location.respawn"),

        DOOR_HUNTER("door.hunter"),

        BORDER_MAP("border.map"),
        BORDER_OPGAME("border.opgame"),
        BORDER_HUNTERZONE("border.hunterzone");

        private final String path;

        private PathType(String path) {
            this.path = path;
        }

        public String getPath() {
            return path;
        }
    }
}

File: src/main/kotlin/jp/aoichaan0513/A_TosoGame_Live/API/Manager/World/WorldManager.kt
package jp.aoichaan0513.A_TosoGame_Live.API.Manager.World

import jp.aoichaan0513.A_TosoGame_Live.API.Manager.GameManager
import jp.aoichaan0513.A_TosoGame_Live.Main
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.World

class WorldManager {
    companion object {

        var world: World
            get() = Bukkit.getWorld(worldName)!!
            set(world) {
                if (GameManager.isGame()) return

                Main.mainConfig.set("worldName", world.name)
                Main.setMainConfig()
                Main.pluginInstance.reloadConfig()
                return
            }

        val worldName: String
            get() = Main.mainConfig.getString("worldName")!!

        fun setWorld(world: String) {
            if (GameManager.isGame()) return

            Main.mainConfig.set("worldName", world)
            Main.setMainConfig()
            Main.pluginInstance.reloadConfig()
            return
        }
    }

    enum class Difficulty(val id: Int, val displayName: String, val color: ChatColor) {
        EASY(0, "${ChatColor.GREEN}イージー", ChatColor.GREEN),
        NORMAL(1, "${ChatColor.YELLOW}ノーマル", ChatColor.YELLOW),
        HARD(2, "${ChatColor.RED}ハード", ChatColor.RED),
        HARDCORE(3, "${ChatColor.DARK_RED}ハードコア", ChatColor.DARK_RED);

        val difficultyName
            get() = name.toLowerCase()

        companion object {

            fun getDifficulty(id: Int): Difficulty {
                return values().firstOrNull { it.id == id } ?: NORMAL
            }

            fun getDifficulty(name: String): Difficulty {
                return values().firstOrNull { it.name.toUpperCase().equals(name.toUpperCase()) } ?: NORMAL
            }
        }
    }

    // ゲームタイプ
    enum class GameType {
        START,
        RESPAWN
    }

    enum class PathType(val path: String) {
        MAP("map"),

        GAME_TIME("game.time"),
        GAME_OTHER("game.other"),

        OPGAME_DICE("opgame.dice"),

        LOCATION_OPGAME("location.opgame"),
        LOCATION_GOPGAME("location.gopgame"),
        LOCATION_HUNTER("location.hunter"),
        LOCATION_JAIL("location.jail"),
        LOCATION_RESPAWN("location.respawn"),
        DOOR_HUNTER("door.hunter"),

        BORDER_MAP("border.map"),
        BORDER_OPGAME("border.opgame"),
        BORDER_HUNTERZONE("border.hunterzone");
    }
}